dataset/src/test/scala/frameless/functions/UdfTest.scala

```diff
@@ -0,0 +1,251 @@
+package frameless.functions
+
+import frameless.{TypedDataset, TypedDatasetSuite, TypedEncoder, X3, X5}
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+import frameless.functions.udfs._
+
+class UdfTest extends TypedDatasetSuite {
+
+  test("one argument udf") {
+    def prop[A, B, C](f1: A => A)(data: Vector[X3[A, B, C]])(
```

Comment:
What about moving implicits to type parameters? By the way, if we have implicit encoders for `A`,`B`,`C`, it should be derived automatically for `X3`, so no need to require it.